src/PowerShellEditorServices/Session/PowerShellContext.cs

```diff
@@ -1113,30 +1113,143 @@ public async Task SetWorkingDirectory(string path, bool isPathAlreadyEscaped)
             {
                 if (!isPathAlreadyEscaped)
                 {
-                    path = EscapePath(path, false);
+                    path = GlobEscapePath(path);
                 }
 
                 runspaceHandle.Runspace.SessionStateProxy.Path.SetLocation(path);
             }
         }
 
+        /// <summary>
+        /// Fully escape a given path for use in PowerShell script.
+        /// Note: this will not work with PowerShell.AddParameter()
+        /// </summary>
+        /// <param name="path">The path to escape.</param>
+        /// <returns>An escaped version of the path that can be embedded in PowerShell script.</returns>
+        internal static string FullyPowerShellEscapePath(string path)
+        {
+            string globEscapedPath = GlobEscapePath(path);
```

Comment:
A command invocation doesn't support wildcards, so escaping pattern chars will cause the file not to be found in dot source expressions.

```powershell
New-Item ./[0].ps1
. './`[0`].ps1'

# . : The term '.\`[0`].ps1' is not recognized as the name of a cmdlet, function, script file, or operable program. Check
# the spelling of the name, or if a path was included, verify that the path is correct and try again.
# At line:1 char:3
# + . '.\`[0`].ps1'
# +   ~~~~~~~~~~~~
#     + CategoryInfo          : ObjectNotFound: (.\`[0].ps1:String) [], CommandNotFoundException
#     + FullyQualifiedErrorId : CommandNotFoundException
```

Reply:
Ah, I knew separating out the quoting method would pay dividends!